tests/apollo/test_skvbc_dbsnapshot.py

```diff
@@ -846,7 +833,51 @@ async def test_signed_state_snapshot_read_as_of_req_non_existent_checkpoint(self
         for ser_rsi in ser_rsis.values():
             rsi_resp = cmf_msgs.ReconfigurationResponse.deserialize(ser_rsi)[0]
             self.assertEqual(rsi_resp.response.status, cmf_msgs.SnapshotResponseStatus.SnapshotNonExistent)
-    
+
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n == 7)
+    @verify_linearizability()
+    async def test_restore_from_snapshot_of_other(self, bft_network, tracker):
+        initial_prim = 0
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
+        for i in range(DB_CHECKPOINT_WIN_SIZE):  # get to a checkpoint
+            await skvbc.send_write_kv_set()
```

Comment:
There is a function in skvbc_commit_path called send_kvs_sequentially which does the same and logs the index of every sent kv.
You can move it to the ApolloTest class and call it using self.send_kvs_sequentially.

Another thing (which might be ever clearer) is to add a function of your own send_until_checkpoints_reached(checkpoint_count) which calls send_kvs_sequentially.

Reply:
thanks.
Used send_kvs_sequentially only, as sometimes we don't send kvs until a checkpoint